src/crypto/openssl/aead.rs

```diff
@@ -190,11 +190,14 @@ impl CipherSuite {
         let mut signer = openssl::sign::Signer::new(openssl::hash::MessageDigest::sha256(), &key)?;
 
         // for current platforms there is no issue casting from usize to u64
-        let aad_len = (aad.len() as u64).to_be_bytes();
-        let ct_len = (encrypted.len() as u64).to_be_bytes();
-        for buf in [&aad_len, &ct_len, nonce, aad, encrypted] {
+        let aad_len = &(aad.len() as u64).to_be_bytes();
+        let ct_len = &(encrypted.len() as u64).to_be_bytes();
+        let tag_len = &(self.auth_tag_len as u64).to_be_bytes();
+
+        for buf in [aad_len, ct_len, tag_len, nonce, aad, encrypted] {
```

Comment:
loop needed?

Reply:
I double checked, and there is no other way than to allocate another buffer then. Openssl requires continous slice. However `encrypted` can be quite large as it holds the entire encrypted frame. Thus I would refrain from another allocation and just keep the loop.

P.s. I also tried an intermediate solution only copying 
`aad_len, ct_len, tag_len, nonce, aad` into a new buffer to reduce the amount of function calls. However I could see no real difference in the benchmarks at least on my system. At some point we might want to come back to testing WASM performance, not sure how it behaves there. But that is whole topic for it self ...